modules/gapi/test/gapi_statefulkernel_tests.cpp

```diff
@@ -0,0 +1,201 @@
+namespace
+{
+    // G_TYPED_KERNEL(GHasSourceChanged, <cv::GOpaque<bool>(GMat)>, "org.opencv.test.has_souce_changed")
+    // {
+    //     static GOpaqueDesc outMeta(GMatDesc /* in */) { return empty_gopaque_desc(); }
+    // };
+
+    // GAPI_OCV_STATEFUL_KERNEL(GOCVHasSourceChanged, GHasSourceChanged, cv::Mat)
+    // {
+    //     static void setup(const cv::GMatDesc &in, std::shared_ptr<cv::Mat> &state)
+    //     {
+    //         state.reset(new cv::Mat{  });
+    //         in.copyTo(*state);
+    //     }
+
+    //     static void run(const cv::Mat &in, bool &out, cv::Mat& state)
+    //     {
+    //         // runtime initState is acttually here
+    //         if (mat.empty())
+    //         {
+
+    //         }
+    //         out = (cv::norm(in, state, NORM_INF) != 0);
+    //     }
+    // };
+
+    G_TYPED_KERNEL(GCountCalls, <cv::GOpaque<int>(GMat)>, "org.opencv.test.count_calls")
+    {
+        static GOpaqueDesc outMeta(GMatDesc /* in */) { return empty_gopaque_desc(); }
+    };
+
+    GAPI_OCV_STATEFUL_KERNEL(GOCVCountCalls, GCountCalls, int)
+    {
+        static void setup(const cv::GMatDesc &/* in */, std::shared_ptr<int> &state)
+        {
+            state.reset(new int{  });
+        }
+
+        static void run(const cv::Mat &/* in */, int &out, int& state)
+        {
+            out = ++state;
+        }
+    };
+};
+
+// TEST(InitState, InitStateIsCalledOnce)
+// {
+//     cv::Size imageSize(640, 480);
+//     cv::Mat sourceImage(imageSize, CV_8UC1),
+//             differentImage(imageSize, CV_8UC1);
+//     cv::randu(sourceImage, cv::Scalar(0), cv::Scalar(255));
+//     cv::randu(differentImage, cv::Scalar(0), cv::Scalar(255));
+
+//     std::vector<cv::Mat> stream(3, sourceImage);
+
+//     bool isChanged = true;
+
+//     GMat in;
+//     auto out = GHasSourceChanged::on(in);
+//     cv::GComputation comp(cv::GIn(in), cv::GOut(out));
+
+//     const auto pkg = cv::gapi::kernels<GOCVHasSourceChanged>();
+//     comp.apply(cv::gin(stream[0]), cv::gout(isChanged), cv::compile_args(pkg));
+//     EXPECT_FALSE(isChanged);
+
+//     for (const auto &frame: stream)
+//     {
+//         comp.apply(cv::gin(frame), cv::gout(isChanged));
+//         EXPECT_FALSE(isChanged);
+//     }
+
+//     comp.apply(cv::gin(differentImage), cv::gout(isChanged));
+//     EXPECT_TRUE(isChanged);  
+//}
+
+TEST(StatefuKernel, StateIsMutableInRuntime)
+{ 
+    constexpr int expectedCallsCount = 10;
+
+    //graph with empty mat shouldn't be compiled - TO CREATE A TICKET 
+    cv::Mat dummyIn;
+    // to init in other tests
+    int actualCallsCount = 0;
+
+    // Declaration of G-API expression
+    GMat in;
+    GOpaque<int> out = GCountCalls::on(in);
+    cv::GComputation comp(cv::GIn(in), cv::GOut(out));
+
+    const auto pkg = cv::gapi::kernels<GOCVCountCalls>();
+
+    // Compilation of G-API expression
+    auto callsCounter = comp.compile(cv::descr_of(dummyIn), cv::compile_args(pkg));
+
+    // Streaming
```

Comment:
You and me understand what do you mean by `Streaming` here. Others won't.

Reply:
Fixed.